Client crash on invalid mod

**Describe the bug**
User has mod installed with a bad file path and the client crashes on start up.

**To Reproduce**
Hard to say but I guess put some : in the path to the icon. But fix should be able without a way to reproduce.

**Log or error message**
This is my log file:
```
java.lang.RuntimeException: Exception in Application init method
	at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:895)
	at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(LauncherImpl.java:195)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'fileOpeningHandler' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/os/FileOpeningHandler.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'replayService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/replay/ReplayService.class]: Unsatisfied dependency expressed through constructor parameter 5; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'gameService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/game/GameService.class]: Unsatisfied dependency expressed through constructor parameter 5; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'featuredModUpdaterConfig' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/config/FeaturedModUpdaterConfig.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'modService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/mod/ModService.class]: Invocation of init method failed; nested exception is java.nio.file.InvalidPathException: Illegal char <:> at index 5: table: 505ad397
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:797)
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:227)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1358)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1204)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:893)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:879)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:551)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:758)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:750)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:397)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315)
	at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:140)
	at com.faforever.client.FafClientApplication.init(FafClientApplication.java:97)
	at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:824)
	... 2 more
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'replayService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/replay/ReplayService.class]: Unsatisfied dependency expressed through constructor parameter 5; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'gameService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/game/GameService.class]: Unsatisfied dependency expressed through constructor parameter 5; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'featuredModUpdaterConfig' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/config/FeaturedModUpdaterConfig.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'modService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/mod/ModService.class]: Invocation of init method failed; nested exception is java.nio.file.InvalidPathException: Illegal char <:> at index 5: table: 505ad397
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:797)
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:227)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1358)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1204)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1304)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1224)
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:884)
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:788)
	... 21 more
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'gameService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/game/GameService.class]: Unsatisfied dependency expressed through constructor parameter 5; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'featuredModUpdaterConfig' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/config/FeaturedModUpdaterConfig.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'modService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/mod/ModService.class]: Invocation of init method failed; nested exception is java.nio.file.InvalidPathException: Illegal char <:> at index 5: table: 505ad397
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:797)
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:227)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1358)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1204)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1304)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1224)
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:884)
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:788)
	... 35 more
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'featuredModUpdaterConfig' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/config/FeaturedModUpdaterConfig.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'modService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/mod/ModService.class]: Invocation of init method failed; nested exception is java.nio.file.InvalidPathException: Illegal char <:> at index 5: table: 505ad397
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:797)
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:227)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1358)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1204)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:408)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1338)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1177)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1304)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1224)
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:884)
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:788)
	... 49 more
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'modService' defined in URL [jar:file:/C:/Program%20Files/Downlord's%20FAF%20Client/lib/downlords-faf-client-1.2.0.jar!/com/faforever/client/mod/ModService.class]: Invocation of init method failed; nested exception is java.nio.file.InvalidPathException: Illegal char <:> at index 5: table: 505ad397
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1796)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:595)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1304)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1224)
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:884)
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:788)
	... 72 more
Caused by: java.nio.file.InvalidPathException: Illegal char <:> at index 5: table: 505ad397
	at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
	at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:229)
	at java.base/java.nio.file.Path.of(Path.java:147)
	at java.base/java.nio.file.Paths.get(Paths.java:69)
	at com.faforever.client.mod.ModVersion.lambda$fromModInfo$1(ModVersion.java:110)
	at java.base/java.util.Optional.map(Optional.java:258)
	at com.faforever.client.mod.ModVersion.fromModInfo(ModVersion.java:110)
	at com.faforever.client.mod.ModService.extractModInfo(ModService.java:267)
	at com.faforever.client.mod.ModService.extractModInfo(ModService.java:261)
	at com.faforever.client.mod.ModService.addMod(ModService.java:414)
	at com.faforever.client.mod.ModService.loadInstalledMods(ModService.java:153)
	at com.faforever.client.mod.ModService.onModDirectoryReady(ModService.java:126)
	at com.faforever.client.mod.ModService.lambda$afterPropertiesSet$0(ModService.java:110)
	at com.faforever.client.mod.ModService.afterPropertiesSet(ModService.java:113)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1855)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1792)
	... 83 more
```
**Expected behavior**
The client moves the mod to the corrupted folder, or just does not import this very mod. Also logs the behaviour. I think there is a message and special behaviour for corrupted mods just that it does not apply here for some reason.

**OS**
Windows

**Wanna have the bug fixed quickly?**
Visit Issue hunt...
[![Issue hunt](https://github.com/BoostIO/issuehunt-materials/raw/master/v1/issuehunt-button-v1.svg?sanitize=true)](https://issuehunt.io/r/FAForever/downlords-faf-client)

--- a/src/main/java/com/faforever/client/chat/ChannelTabController.java
+++ b/src/main/java/com/faforever/client/chat/ChannelTabController.java
@@ -146,6 +146,7 @@ public class ChannelTabController extends AbstractChatTabController {
   public ListView<CategoryOrChatUserListItem> chatUserListView;
   public VBox topicPane;
   public TextFlow topicText;
+  public ToggleButton toggleSidePaneButton;
 
   private Channel channel;
   private Popup filterUserPopup;
@@ -299,6 +300,19 @@ public class ChannelTabController extends AbstractChatTabController {
     chatUserListView.setCellFactory(param -> new ChatUserListCell(uiService));
 
     autoCompletionHelper.bindTo(messageTextField());
+
+    initializeSideToggle();
+  }
+
+  private void initializeSideToggle() {
+    toggleSidePaneButton.setSelected(preferencesService.getPreferences().getChat().isPlayerListShown());
+    JavaFxUtil.bind(channelTabScrollPaneVBox.visibleProperty(), toggleSidePaneButton.selectedProperty());
+    JavaFxUtil.bind(channelTabScrollPaneVBox.managedProperty(), channelTabScrollPaneVBox.visibleProperty());
+    JavaFxUtil.addListener(toggleSidePaneButton.selectedProperty(), (observable, oldValue, newValue) -> splitPane.setDividerPositions(newValue ? 0.8 : 1));
+    JavaFxUtil.addListener(toggleSidePaneButton.selectedProperty(), (observable, oldValue, newValue) -> {
+      preferencesService.getPreferences().getChat().setPlayerListShown(newValue);
+      preferencesService.storeInBackground();
+    });
   }
 
   @Override
@@ -404,7 +418,7 @@ public class ChannelTabController extends AbstractChatTabController {
     //Workaround for issue #1080 https://github.com/FAForever/downlords-faf-client/issues/1080
     Platform.runLater(() -> {
       try {
-        engine.executeScript("removeUserMessageClass(\'" + String.format(USER_CSS_CLASS_FORMAT, chatUser.getUsername()) + "\',\'" + cssClass + "\');");
+        engine.executeScript("removeUserMessageClass('" + String.format(USER_CSS_CLASS_FORMAT, chatUser.getUsername()) + "','" + cssClass + "');");
       } catch (Exception ignored) {
         //before with "getJsObject().call..." if the engine was not yet loaded the Exception was ignored and hence I know to the same
         //TODO: only accept calls after the engine loaded the page completely

--- a/src/main/java/com/faforever/client/game/CustomGamesController.java
+++ b/src/main/java/com/faforever/client/game/CustomGamesController.java
@@ -18,7 +18,6 @@ import javafx.beans.value.ChangeListener;
 import javafx.beans.value.WeakChangeListener;
 import javafx.collections.ObservableList;
 import javafx.collections.transformation.FilteredList;
-import javafx.css.PseudoClass;
 import javafx.scene.Node;
 import javafx.scene.control.Button;
 import javafx.scene.control.CheckBox;
@@ -59,12 +58,6 @@ public class CustomGamesController extends AbstractViewController<Node> {
   private static final Predicate<Game> OPEN_CUSTOM_GAMES_PREDICATE = gameInfoBean ->
       gameInfoBean.getStatus() == GameStatus.OPEN
           && !HIDDEN_FEATURED_MODS.contains(gameInfoBean.getFeaturedMod());
-  @VisibleForTesting
-  static final PseudoClass PSEUDO_CLASS_RIGHT = PseudoClass.getPseudoClass("right");
-  @VisibleForTesting
-  static final PseudoClass PSEUDO_CLASS_OPENED = PseudoClass.getPseudoClass("opened");
-  @VisibleForTesting
-  static final PseudoClass PSEUDO_CLASS_CLOSED = PseudoClass.getPseudoClass("closed");
 
   private final UiService uiService;
   private final GameService gameService;
@@ -163,22 +156,13 @@ public class CustomGamesController extends AbstractViewController<Node> {
 
     setSelectedGame(null);
 
-    toggleGameDetailPaneButton.pseudoClassStateChanged(PSEUDO_CLASS_RIGHT, true);
     toggleGameDetailPaneButton.selectedProperty().addListener(observable -> {
-      updateGameDetailPaneButton();
       preferencesService.getPreferences().setShowGameDetailsSidePane(toggleGameDetailPaneButton.isSelected());
       preferencesService.storeInBackground();
     });
     toggleGameDetailPaneButton.setSelected(preferencesService.getPreferences().isShowGameDetailsSidePane());
-    updateGameDetailPaneButton();
 
     eventBus.register(this);
-  }
-
-  private void updateGameDetailPaneButton() {
-    boolean selected = toggleGameDetailPaneButton.isSelected();
-    toggleGameDetailPaneButton.pseudoClassStateChanged(PSEUDO_CLASS_CLOSED, !selected);
-    toggleGameDetailPaneButton.pseudoClassStateChanged(PSEUDO_CLASS_OPENED, selected);
   }
 
   @Override

--- a/src/main/java/com/faforever/client/preferences/ChatPrefs.java
+++ b/src/main/java/com/faforever/client/preferences/ChatPrefs.java
@@ -4,7 +4,6 @@ import com.faforever.client.chat.ChatColorMode;
 import com.faforever.client.chat.ChatFormat;
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.ImmutableMap;
-
 import javafx.beans.property.BooleanProperty;
 import javafx.beans.property.DoubleProperty;
 import javafx.beans.property.IntegerProperty;
@@ -26,7 +25,9 @@ import java.util.Locale;
 import java.util.Optional;
 
 import static com.faforever.client.chat.ChatColorMode.CUSTOM;
-import static com.faforever.client.preferences.LanguageChannel.*;
+import static com.faforever.client.preferences.LanguageChannel.FRENCH;
+import static com.faforever.client.preferences.LanguageChannel.GERMAN;
+import static com.faforever.client.preferences.LanguageChannel.RUSSIAN;
 
 public class ChatPrefs {
 
@@ -46,6 +47,7 @@ public class ChatPrefs {
   private final IntegerProperty channelTabScrollPaneWidth;
   private final MapProperty<String, Color> userToColor;
   private final BooleanProperty hideFoeMessages;
+  private final BooleanProperty playerListShown;
   private final ObjectProperty<TimeInfo> timeFormat;
   private final ObjectProperty<ChatFormat> chatFormat;
   private final ListProperty<String> autoJoinChannels;
@@ -67,6 +69,7 @@ public class ChatPrefs {
     idleThreshold = new SimpleIntegerProperty(10);
     chatFormat = new SimpleObjectProperty<>(ChatFormat.COMPACT);
     autoJoinChannels = new SimpleListProperty<>(FXCollections.observableArrayList());
+    playerListShown = new SimpleBooleanProperty(true);
 
     Locale localeLanguage = new Locale(Locale.getDefault().getLanguage());
     Optional.ofNullable(LOCALE_LANGUAGES_TO_CHANNELS.get(localeLanguage))
@@ -204,5 +207,17 @@ public class ChatPrefs {
 
   public ObservableList<String> getAutoJoinChannels() {
     return autoJoinChannels.get();
+  }
+
+  public boolean isPlayerListShown() {
+    return playerListShown.get();
+  }
+
+  public void setPlayerListShown(boolean playerListShown) {
+    this.playerListShown.set(playerListShown);
+  }
+
+  public BooleanProperty playerListShownProperty() {
+    return playerListShown;
   }
 }

--- a/src/main/resources/theme/chat/channel_tab.fxml
+++ b/src/main/resources/theme/chat/channel_tab.fxml
@@ -15,7 +15,6 @@
 <?import javafx.scene.layout.VBox?>
 <?import javafx.scene.text.TextFlow?>
 <?import javafx.scene.web.WebView?>
-
 <Tab fx:id="channelTabRoot" xmlns="http://javafx.com/javafx/11.0.1" xmlns:fx="http://javafx.com/fxml/1" fx:controller="com.faforever.client.chat.ChannelTabController">
   <VBox maxHeight="1.7976931348623157E308" maxWidth="1.7976931348623157E308" minHeight="0.0" minWidth="0.0" onKeyReleased="#onKeyReleased">
     <children>
@@ -33,45 +32,63 @@
                           <WebView fx:id="messagesWebView" minHeight="100.0" minWidth="100.0" prefHeight="-1.0" prefWidth="-1.0" AnchorPane.bottomAnchor="0.0" AnchorPane.leftAnchor="0.0" AnchorPane.rightAnchor="0.0" AnchorPane.topAnchor="0.0" />
                           <HBox fx:id="searchFieldContainer" visible="false" AnchorPane.rightAnchor="0.0" AnchorPane.topAnchor="0.0">
                               <children>
-                                  <Button fx:id="closeSearchFieldButton" focusTraversable="false" minHeight="-Infinity" minWidth="-Infinity" onAction="#onSearchFieldClose" styleClass="close-search-text-button">
+                                  <Button fx:id="closeSearchFieldButton" focusTraversable="false" minHeight="-Infinity"
+                                          minWidth="-Infinity" onAction="#onSearchFieldClose"
+                                          styleClass="close-search-text-button">
                                       <graphic>
                                           <StackPane>
-                                              <Region minHeight="-Infinity" minWidth="-Infinity" styleClass="icon,close-icon" />
+                                              <Region minHeight="-Infinity" minWidth="-Infinity"
+                                                      styleClass="icon,close-icon"/>
                                           </StackPane>
                                       </graphic>
                                   </Button>
-                                  <TextField fx:id="searchField" promptText="%chat.filter.prompt" visible="false" />
+                                  <TextField fx:id="searchField" promptText="%chat.filter.prompt" visible="false"/>
                               </children>
                           </HBox>
+                          <ToggleButton fx:id="toggleSidePaneButton" selected="true" mnemonicParsing="false"
+                                        styleClass="pane-toggler, right" AnchorPane.rightAnchor="0.0"
+                                        AnchorPane.topAnchor="40.0">
+                              <graphic>
+                                  <Region styleClass="icon,toggle-icon"/>
+                              </graphic>
+                          </ToggleButton>
                       </children>
                   </AnchorPane>
-                    <TextField fx:id="messageTextField" maxWidth="1.7976931348623157E308" onAction="#onSendMessage" promptText="%chat.messagePrompt">
-                    </TextField>
+                   <TextField fx:id="messageTextField" maxWidth="1.7976931348623157E308" onAction="#onSendMessage"
+                              promptText="%chat.messagePrompt">
+                   </TextField>
                </children>
             </VBox>
 
-          <VBox fx:id="channelTabScrollPaneVBox" maxHeight="1.7976931348623157E308" styleClass="chat-channel-users-pane" SplitPane.resizableWithParent="false">
-            <children>
-              <HBox alignment="CENTER" spacing="10.0">
+            <VBox fx:id="channelTabScrollPaneVBox" maxHeight="1.7976931348623157E308"
+                  styleClass="chat-channel-users-pane" SplitPane.resizableWithParent="false" minWidth="300.0">
                 <children>
-                    <TextField fx:id="userSearchTextField" maxWidth="1.7976931348623157E308" minWidth="10.0" prefColumnCount="5" HBox.hgrow="ALWAYS" />
-                    <ToggleButton fx:id="advancedUserFilter" mnemonicParsing="false" onAction="#onAdvancedUserFilter" HBox.hgrow="NEVER">
-                        <graphic>
-                            <Region styleClass="icon,filter-icon" />
-                        </graphic>
-                    </ToggleButton>
-                </children>
+                    <HBox alignment="CENTER" spacing="10.0">
+                        <children>
+                            <TextField fx:id="userSearchTextField" maxWidth="1.7976931348623157E308" minWidth="10.0"
+                                       prefColumnCount="5" HBox.hgrow="ALWAYS"/>
+                            <ToggleButton fx:id="advancedUserFilter" mnemonicParsing="false"
+                                          onAction="#onAdvancedUserFilter" HBox.hgrow="NEVER">
+                                <graphic>
+                                    <Region styleClass="icon,filter-icon"/>
+                                </graphic>
+                            </ToggleButton>
+                        </children>
                         <VBox.margin>
-                           <Insets left="10.0" right="10.0" />
+                            <Insets left="10.0" right="10.0"/>
                         </VBox.margin>
-              </HBox>
-              <AnchorPane maxHeight="1.7976931348623157E308" maxWidth="1.7976931348623157E308" VBox.vgrow="ALWAYS">
-                    <children>
-                        <ListView fx:id="chatUserListView" maxHeight="1.7976931348623157E308" maxWidth="1.7976931348623157E308" styleClass="chat-user-list-view" AnchorPane.bottomAnchor="0.0" AnchorPane.leftAnchor="0.0" AnchorPane.rightAnchor="0.0" AnchorPane.topAnchor="0.0">
-                        </ListView>
-                    </children>
-              </AnchorPane>
-            </children>
+                    </HBox>
+                    <AnchorPane maxHeight="1.7976931348623157E308" maxWidth="1.7976931348623157E308"
+                                VBox.vgrow="ALWAYS">
+                        <children>
+                            <ListView fx:id="chatUserListView" maxHeight="1.7976931348623157E308"
+                                      maxWidth="1.7976931348623157E308" styleClass="chat-user-list-view"
+                                      AnchorPane.bottomAnchor="0.0" AnchorPane.leftAnchor="0.0"
+                                      AnchorPane.rightAnchor="0.0" AnchorPane.topAnchor="0.0">
+                            </ListView>
+                        </children>
+                    </AnchorPane>
+                </children>
           </VBox>
         </SplitPane>
     </children>

--- a/src/main/resources/theme/play/custom_games.fxml
+++ b/src/main/resources/theme/play/custom_games.fxml
@@ -32,22 +32,28 @@
                                               <HBox alignment="BASELINE_LEFT" />
                                           </children>
                                       </FlowPane>
-                                      <ToggleButton fx:id="tableButton" minWidth="-Infinity" mnemonicParsing="false" onAction="#onTableButtonClicked" styleClass="game-list-type" text="%view.table">
+                                      <ToggleButton fx:id="tableButton" minWidth="-Infinity" mnemonicParsing="false"
+                                                    onAction="#onTableButtonClicked" styleClass="game-list-type"
+                                                    text="%view.table">
                                           <toggleGroup>
-                                              <ToggleGroup fx:id="viewToggleGroup" />
+                                              <ToggleGroup fx:id="viewToggleGroup"/>
                                           </toggleGroup>
                                           <graphic>
-                                              <Region styleClass="icon,table-icon" />
+                                              <Region styleClass="icon,table-icon"/>
                                           </graphic>
                                       </ToggleButton>
-                                      <ToggleButton fx:id="tilesButton" minWidth="-Infinity" mnemonicParsing="false" onAction="#onTilesButtonClicked" selected="true" styleClass="game-list-type" text="%view.tiles" toggleGroup="$viewToggleGroup">
+                                      <ToggleButton fx:id="tilesButton" minWidth="-Infinity" mnemonicParsing="false"
+                                                    onAction="#onTilesButtonClicked" selected="true"
+                                                    styleClass="game-list-type" text="%view.tiles"
+                                                    toggleGroup="$viewToggleGroup">
                                           <graphic>
-                                              <Region styleClass="icon,tiles-icon" />
+                                              <Region styleClass="icon,tiles-icon"/>
                                           </graphic>
                                       </ToggleButton>
-                                      <ToggleButton fx:id="toggleGameDetailPaneButton" mnemonicParsing="false" styleClass="pane-toggler">
+                                      <ToggleButton fx:id="toggleGameDetailPaneButton" mnemonicParsing="false"
+                                                    styleClass="pane-toggler, right">
                                           <graphic>
-                                              <Region styleClass="icon,toggle-icon" />
+                                              <Region styleClass="icon,toggle-icon"/>
                                           </graphic>
                                       </ToggleButton>
                                   </children>

--- a/src/main/resources/theme/style.css
+++ b/src/main/resources/theme/style.css
@@ -737,13 +737,13 @@
   -fx-padding: 0.833333em .25em;
 }
 
-.toggle-button.pane-toggler:left:closed .toggle-icon,
-.toggle-button.pane-toggler:right:opened .toggle-icon {
+.toggle-button.pane-toggler.left .toggle-icon,
+.toggle-button.pane-toggler.right:selected .toggle-icon {
   -fx-shape: "M11.611,10.049l-4.76-4.873c-0.303-0.31-0.297-0.804,0.012-1.105c0.309-0.304,0.803-0.293,1.105,0.012l5.306,5.433c0.304,0.31,0.296,0.805-0.012,1.105L7.83,15.928c-0.152,0.148-0.35,0.223-0.547,0.223c-0.203,0-0.406-0.08-0.559-0.236c-0.303-0.309-0.295-0.803,0.012-1.104L11.611,10.049z";
 }
 
-.toggle-button.pane-toggler:left:opened .toggle-icon,
-.toggle-button.pane-toggler:right:closed .toggle-icon {
+.toggle-button.pane-toggler.left:selected .toggle-icon,
+.toggle-button.pane-toggler.right .toggle-icon {
   -fx-shape: "M8.388,10.049l4.76-4.873c0.303-0.31,0.297-0.804-0.012-1.105c-0.309-0.304-0.803-0.293-1.105,0.012L6.726,9.516c-0.303,0.31-0.296,0.805,0.012,1.105l5.433,5.307c0.152,0.148,0.35,0.223,0.547,0.223c0.203,0,0.406-0.08,0.559-0.236c0.303-0.309,0.295-0.803-0.012-1.104L8.388,10.049z";
 }
 
@@ -757,7 +757,7 @@
   -fx-padding: 0 0.25em;
 }
 
-.pane-toggler:right {
+.pane-toggler.right {
   -fx-border-width: 0;
   -fx-background-radius: 3 0 0 3;
 }

--- a/src/test/java/com/faforever/client/chat/ChannelTabControllerTest.java
+++ b/src/test/java/com/faforever/client/chat/ChannelTabControllerTest.java
@@ -49,6 +49,7 @@ import static org.junit.Assert.assertTrue;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.anyString;
 import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.atLeast;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.verify;
@@ -388,5 +389,25 @@ public class ChannelTabControllerTest extends AbstractPlainJavaFxTest {
     defaultChannel.setTopic("topic2: https://faforever.com topic3: https://faforever.com/example");
     WaitForAsyncUtils.waitForFxEvents();
     assertEquals(instance.topicText.getChildren().size(), 4);
+  }
+
+  @Test
+  public void testChannelListHide() {
+    instance.toggleSidePaneButton.fire();
+    WaitForAsyncUtils.waitForFxEvents();
+    assertFalse(instance.channelTabScrollPaneVBox.isManaged());
+    assertFalse(instance.toggleSidePaneButton.isSelected());
+  }
+
+  @Test
+  public void testHideSidePane() {
+    instance.toggleSidePaneButton.fire();
+    WaitForAsyncUtils.waitForFxEvents();
+
+    assertFalse(preferencesService.getPreferences().getChat().isPlayerListShown());
+    verify(preferencesService, atLeast(1)).storeInBackground();
+
+    assertFalse(instance.channelTabScrollPaneVBox.isManaged());
+    assertFalse(instance.channelTabScrollPaneVBox.isVisible());
   }
 }

--- a/src/test/java/com/faforever/client/game/CustomGamesControllerTest.java
+++ b/src/test/java/com/faforever/client/game/CustomGamesControllerTest.java
@@ -20,8 +20,6 @@ import org.mockito.Mock;
 import org.mockito.junit.MockitoJUnitRunner;
 import org.testfx.util.WaitForAsyncUtils;
 
-import static org.hamcrest.CoreMatchers.hasItem;
-import static org.hamcrest.MatcherAssert.assertThat;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
@@ -154,7 +152,6 @@ public class CustomGamesControllerTest extends AbstractPlainJavaFxTest {
     assertFalse(preferencesService.getPreferences().isShowGameDetailsSidePane());
     verify(preferencesService, atLeast(2)).storeInBackground();
 
-    assertThat(instance.toggleGameDetailPaneButton.getPseudoClassStates(), hasItem(CustomGamesController.PSEUDO_CLASS_CLOSED));
     assertFalse(instance.gameDetailPane.isManaged());
     assertFalse(instance.gameDetailPane.isVisible());
   }

